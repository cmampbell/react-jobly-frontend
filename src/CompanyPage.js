import React from 'react';
import { useLoaderData } from 'react-router-dom';

import JobCard from './JobCard';

const CompanyPage = () => {
    const company = useLoaderData()
    return (
        <div className='CompanyPage'>
            <h1>{company.name}</h1>
            <p>{company.description}</p>
            {company.jobs.map((job) => <JobCard job={job} key={job.id}/>)}
        </div>
    )
}

export default CompanyPage;