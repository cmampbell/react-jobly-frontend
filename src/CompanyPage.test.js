import { render } from '@testing-library/react';
import CompanyPage from './CompanyPage';
import { RouterProvider, createMemoryRouter, Route } from 'react-router-dom';

const testCompany = {
  handle: 'test1',
  name: 'Name1',
  numEmployees: 45,
  description: 'Test company 1',
  jobs: [{id: 1, title: 'test1', companyName: 'test', salary: '50000', equity: '.03'},
          {id: 2, title: 'test2', companyName: 'test2', salary: '500', equity: '.05'}]
}


const route = [
  {
    path: "/",
    element: <CompanyPage />,
    loader: ()=> testCompany
  },
];

const router = createMemoryRouter(route)

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useOutletContext: function () {
    return {
      applyToJob: function () {
        return null
      },
      applications: new Set()
    }
  }
})
);

describe('Smoke and snapshot CompanyPage test', () => {
  
  it('should render without crashing', () => {
    render(<RouterProvider router={router}>
        <CompanyPage />
    </RouterProvider>);
  });

  it('should match snapshot', async () => {
    const { asFragment, findByText } = render(<RouterProvider router={router}><CompanyPage /></RouterProvider>)
    expect(asFragment()).toMatchSnapshot();
    expect(await findByText('Name1')).toBeInTheDocument()
  })
})
