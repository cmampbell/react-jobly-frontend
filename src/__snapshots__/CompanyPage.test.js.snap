// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot CompanyPage test should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyPage"
  >
    <h1>
      Name1
    </h1>
    <p>
      Test company 1
    </p>
    <div
      class="text-start card border-info"
      style="width: 80vw;"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          test1 at test
        </h5>
        <p
          class="card-text"
        >
          Salary: 50000
        </p>
        <p
          class="card-text"
        >
          Equity: .03
        </p>
        <button
          class="btn btn-secondary"
          type="button"
        >
           Submit Application! 
        </button>
      </div>
    </div>
    <div
      class="text-start card border-info"
      style="width: 80vw;"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          test2 at test2
        </h5>
        <p
          class="card-text"
        >
          Salary: 500
        </p>
        <p
          class="card-text"
        >
          Equity: .05
        </p>
        <button
          class="btn btn-secondary"
          type="button"
        >
           Submit Application! 
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
